apps/wfo-ui-surf/components/WfoServiceTicketDetailPage/WfoSubscriptionImpactTable.tsx

```diff
@@ -0,0 +1,226 @@
+    'related_customers';
+const SUBSCRIPTION_IMPACT_FIELD_INFORM_CUSTOMERS: keyof ServiceTicketImpactedObject =
+    'related_customers';
+const SUBSCRIPTION_IMPACT_FIELD_IMS_CALCULATED_IMPACT: keyof ServiceTicketImpactedObject =
+    'ims_circuits';
+const SUBSCRIPTION_IMPACT_FIELD_NOC_MANUAL_OVERRIDE: keyof ServiceTicketImpactedObject =
+    'impact_override';
+
+export const WfoSubscriptionImpactTable = () => {
+    const t = useTranslations(
+        'cim.serviceTickets.detail.tabDetails.general.subscriptionImpactTable',
+    );
+    const queryClient = useQueryClient();
+    const data = queryClient.getQueryData([
+        'serviceTickets',
+        'ca5eab65-761a-4235-aac8-048ff1aa0171',
+    ]);
+    const { theme, toSecondaryColor } = useOrchestratorTheme();
+    const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<
+        Record<string, ReactNode>
+    >({});
+
+    const calculateAffectedCustomers = (
+        object: ServiceTicketImpactedObject,
+    ) => {
+        return object.related_customers.length + 1;
+    };
+
+    const renderImsCircuitsImpact = (object: ServiceTicketImpactedObject) => {
+        const ims_circuits = object.ims_circuits;
+        return ims_circuits
+            ? ims_circuits[ims_circuits.length - 1].impact
+            : 'null';
+    };
+
+    const toggleDetails = (object: ServiceTicketImpactedObject) => {
+        const itemIdToExpandedRowMapValues = { ...itemIdToExpandedRowMap };
+        const id = object.subscription_id ? object.subscription_id : '0';
+        console.log(itemIdToExpandedRowMapValues, 'toggleDetails');
+        console.log(object, 'object');
+
+        if (itemIdToExpandedRowMapValues[id]) {
+            delete itemIdToExpandedRowMapValues[id];
+        } else {
+            const { subscription_description } = object;
+
+            const color = subscription_description ? 'success' : 'danger';
+            const label = subscription_description ? 'Online' : 'Offline';
+            const listItems = [
+                {
+                    title: 'Location',
+                    description: subscription_description,
+                },
+                {
+                    title: 'Online',
+                    description: <EuiText color={color}>{label}</EuiText>,
+                },
+            ];
+            itemIdToExpandedRowMapValues[id] = (
+                <EuiDescriptionList listItems={listItems} />
+            );
+        }
+        setItemIdToExpandedRowMap(itemIdToExpandedRowMapValues);
+    };
+
+    const impactTableColumns = {
+        expander: {
+            field: SUBSCRIPTION_IMPACT_FIELD_ID,
+            width: '20',
+            isExpander: true,
+            name: (
+                <EuiScreenReaderOnly>
+                    <span>Expand rows</span>
+                </EuiScreenReaderOnly>
+            ),
+            render: (
+                value: keyof ServiceTicketImpactedObject,
+                object: ServiceTicketImpactedObject,
```

Comment:
In addition to my comment above: when the type is applied there, you also won't need to specify the type here and in all other render props